pkg/syncer/spec/spec_controller.go

```diff
@@ -96,7 +99,56 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, upstreamClusterName logic
 				c.AddToQueue(gvr, obj)
 			},
 		})
-		klog.InfoS("Set up informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+		klog.InfoS("Set up upstream informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+
+		downstreamInformers.ForResource(gvr).Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
+				unstrob := obj.(*unstructured.Unstructured)
+				klog.InfoS("downstream delete event", "gvr", gvr, "namespace", unstrob.GetNamespace(), "name", unstrob.GetName())
+				//if there is no label then it is not owned by kcp
+				if unstrob.GetLabels() == nil {
+					return
+				}
+				//Use namespace lister
+				namespaces := downstreamClient.Resource(schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "namespaces",
+				})
+				ns, err := namespaces.Get(context.TODO(), unstrob.GetNamespace(), metav1.GetOptions{})
```

Comment:
Please use the appropriate context, not `context.TODO()`

Reply:
Refactor the method to have one? `context.TODO` should have no place in merged production code :)

Reply:
Not possible. This is an informer event handler

Reply:
It'll go away if we switch to the lister 😛

Reply:
Why exactly do we user any lister or client at all? Why not just work completely with keys. It is shorter, less error-phrone, and it matches the logic of the status syncer.

Reply:
We need the ns